drivers/adc/adc_dw.c

```diff
@@ -51,6 +54,19 @@
 #endif
 static void adc_config_irq(void);
 
+
+#ifdef CONFIG_ADC_DW
```

Comment:
why are you checking for CONFIG_ADC_DW here? This driver is built when CONFIG_ADC_DW is defined, so this becomes useless.